### PR TITLE
Add option --audio-channels.

### PR DESCRIPTION
**Pull request by derkallevombau**
When recording audio using ALSA from a mono mic (attached to an USB
soundcard in my case), you get an error message like this:

_[alsa @ 0x5555638680] cannot set channel count to 2 (Invalid argument)_

When using Pulse, it works, but it's a waste of space on your microSD,
USB stick, USB HDD/SDD, network drive or whatever.

This commit implements an option that has the same effect as ffmpeg's
-ac option.

### CODE REVIEW

File: encoder/libav_encoder.cpp [hunk 1]
```diff
@@ -146,9 +146,31 @@ void LibAvEncoder::initAudioInCodec(VideoOptions const *options, StreamInfo cons
...
+		}
+
+		av_dict_free(&format_opts);
+	}
+	else
+	{
+		ret = avformat_open_input(&in_fmt_ctx_, options->audio_device.c_str(), input_fmt, nullptr);
+		if (ret < 0)
+			throw std::runtime_error("libav: cannot open " + options->audio_source + " input device " + options->audio_device);
+	}
```
[the remark this answers is not in the document]
Comment (naushir) [reply encoder/libav_encoder.cpp:173]: Also, I'm not entirely familiar with ``AVDictionary`` types.  There does not seem to be an allocation here, only a free.  Is the former missing?

**Comment by derkallevombau** [comment]
Can you tell me why that Python program failed? What do I have to change?

**Comment by naushir** [comment]
That's because you have a merge commit in your PR.  We don't use merge commits in this repo.

As a final step, could you resubmit this PR with 2 commits - 1) 6812015c816459f388a8147c6e335910efd4e68f and 2) a squash of dd64482def56ff9eb461f29475cdbf6b0b46a713 and ea5541c0744f2f8f71574de0a513e31eb36253fb.

Can you also format your commit messages similar to the other commits in the git log please.  You need a ``Signed-off-by`` tag at the end of the message, and the title of the commit should begin with ``libav:``

This should then be good to merge.

**Comment by derkallevombau** [comment]
Well, I'll give my best ;)

**Comment by derkallevombau** [comment]
Do I have to completely recreate the pr branch, or can I just do a rebase and drop the merge commit?

**Comment by naushir** [comment]
You ought to just be able to drop the merge commit I think.  If you are having problems, let me know!

**Comment by derkallevombau** [comment]
Ok, thank you!

**Pull request merged**